apps/meteor/app/cloud/server/functions/syncWorkspace/fetchWorkspaceSyncPayload.ts

```diff
@@ -0,0 +1,46 @@
+import type { Cloud, Serialized } from '@rocket.chat/core-typings';
+import { serverFetch as fetch } from '@rocket.chat/server-fetch';
+import { v, compile } from 'suretype';
+
+import { CloudWorkspaceConnectionError } from '../../../../../lib/errors/CloudWorkspaceConnectionError';
+import { settings } from '../../../../settings/server';
+
+const workspaceSyncPayloadSchema = v.object({
+	workspaceId: v.string().required(),
+	publicKey: v.string(),
+	license: v.string().required(),
+});
+
+const assertWorkspaceSyncPayload = compile(workspaceSyncPayloadSchema);
+
+export async function fetchWorkspaceSyncPayload({
+	token,
+	data,
+}: {
+	token: string;
+	data: Cloud.WorkspaceSyncRequestPayload;
+}): Promise<Serialized<Cloud.WorkspaceSyncResponse>> {
+	const workspaceRegistrationClientUri = settings.get<string>('Cloud_Workspace_Registration_Client_Uri');
+	const response = await fetch(`${workspaceRegistrationClientUri}/sync`, {
+		method: 'POST',
+		headers: {
+			Authorization: `Bearer ${token}`,
+		},
+		body: data,
+	});
+
+	if (!response.ok) {
+		try {
+			const { error } = await response.json();
+			throw new CloudWorkspaceConnectionError(`Failed to connect to Rocket.Chat Cloud: ${error}`);
```

Comment:
why do you need to throw an error in here if you're ignoring it on the catch? 👀 

maybe you forgot to use the error? or to log it?

Reply:
Huuum im not sure as to why is it built like that, as I only moved to a new file.

But I can definitely look into it and improve it.

Reply:
Pls 🙏🏽